discovery_engine_core/core/src/engine.rs

```diff
@@ -771,10 +811,22 @@ async fn fetch_new_documents_for_stack(
             return Err(Error::StackOpFailed(error));
         }
     };
+    let (documents, errors) = parallel_articles_into_documents(stack.id(), ranker, articles);
+
+    // only return an error if all articles failed
+    if documents.is_empty() && !errors.is_empty() {
+        Err(Error::Errors(errors))
+    } else {
+        Ok(documents)
+    }
+}
 
-    let id = stack.id();
-    let articles_len = articles.len();
-    let (documents, articles_errors) = articles
+fn parallel_articles_into_documents(
```

Comment:
:art: The fact we are doing this in parallel is an implementation detail and I don't think it add anything to the name of the function.